fabtests/functional/unexpected_msg.c

```diff
@@ -234,14 +234,13 @@ static int exchange_unexp_addr(void)
 		/*
 		 * Send two messages - first will be matched to FI_ADDR_UNSPEC
 		 * Second will be matched to directed receive after fi_av_insert
+		 * Message size is not important - so send 1 byte messages
 		 */
-		ret = ft_post_tx_buf(ep, remote_fi_addr, addrlen, 0, &tx_ctx,
-				     tx_buf, mr_desc, ft_tag);
+		ret = (int) ft_tx(ep, remote_fi_addr, 1, &tx_ctx);
```

Comment:
These both need to stay with the ft_post_rx_buf call which don't wait for the completion. ft_tx waits for it to complete which is going to conflict with the get_tx_comp(2) call on line 249 and make the test hang. The goal here is to post 2 sends and then sync with the other side to ensure that they go to the unexpected queue before the receives are posted.